Simplenote/Classes/InterlinkViewController.swift

```diff
@@ -0,0 +1,184 @@
+    }
+}
+
+
+// MARK: - Initialization
+//
+private extension InterlinkViewController {
+
+    func setupRootView() {
+        view.frame = UIScreen.main.bounds
+        view.backgroundColor = .clear
+    }
+
+    func setupBackgroundView() {
+        backgroundView.backgroundColor = .simplenoteAutocompleteBackgroundColor
+    }
+
+    func setupShadowLayer() {
+        let backgroundLayer = backgroundView.layer
+        backgroundLayer.cornerRadius = Metrics.cornerRadius
+        backgroundLayer.shadowRadius = Shadow.radius
+        backgroundLayer.shadowOpacity = Shadow.opacity
+        backgroundLayer.shadowOffset = Shadow.offset
+    }
+
+    func setupTableView() {
+        tableView.register(Value1TableViewCell.self, forCellReuseIdentifier: Value1TableViewCell.reuseIdentifier)
+        tableView.backgroundColor = .clear
+        tableView.separatorColor = .simplenoteDividerColor
+        tableView.tableFooterView = UIView()
+        tableView.layer.cornerRadius = Metrics.cornerRadius
+    }
+}
+
+
+// MARK: - UITableViewDataSource
+//
+extension InterlinkViewController: UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        notes.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let note = notes[indexPath.row]
+        note.ensurePreviewStringsAreAvailable()
+
+        let tableViewCell = tableView.dequeueReusableCell(ofType: Value1TableViewCell.self, for: indexPath)
+        tableViewCell.title = note.titlePreview
+        tableViewCell.backgroundColor = .clear
+        tableViewCell.separatorInset = .zero
+
+        return tableViewCell
+    }
+}
+
+
+// MARK: - Geometry
+//
+private extension InterlinkViewController {
+
+    /// Updates the layout constraints so that the receiver shows up **around** the specified Keyword in the specified TextView
+    ///
+    func refreshConstraints(keywordRange: Range<String.Index>, in textView: UITextView) {
+        let targetHeight = calculateHeight()
+        let targetLocation = calculateLocation(for: targetHeight, around: keywordRange, in: textView)
+        let targetPadding = textView.textContainer.lineFragmentPadding
+
+        tableTopConstraint.constant = targetLocation
+        tableHeightConstraint.constant = targetHeight
+        tableLeadingConstraint.constant = targetPadding
+        tableTrailingConstraint.constant = targetPadding
+    }
+
+    /// Starts tracking ContentOffset changes in our sibling TextView
+    ///
+    func startObservingContentOffset(in textView: UITextView) {
+        kvoOffsetToken = textView.observe(\UITextView.contentOffset, options: [.old, .new]) { [weak self] (_, value) in
+            guard let oldY = value.oldValue?.y, let newY = value.newValue?.y, oldY != newY else {
+                return
+            }
+
+            self?.tableTopConstraint.constant += oldY - newY
+        }
+    }
```

Comment:
I vote for listening for scrollViewDidScroll in the editor and passing it here. KVO gives me chills :-)

Reply:
If it's okay with you, let's address that in a follow up PR! Added to my list of pendings in #914